include/lbann/base.hpp

```diff
@@ -42,6 +42,11 @@ namespace lbann {
 // Forward-declaration.
 class lbann_comm;
 
+// Note that this should only be used to wrap the thing coming out of
+// initialize()! This will be removed when we have proper RAII around
+// these things.
+using world_comm_ptr = std::unique_ptr<lbann_comm, void(*)(lbann_comm*)>;
```

Comment:
The first one is the thing being pointed to, an `lbann_comm` object. The second is the type of the Deleter to be used when the held object gets destroyed. The default is [std::default_delete](https://en.cppreference.com/w/cpp/memory/default_delete). Because this is only used to hold our logical "COMM_WORLD", we want `finalize()` to be called when this thing gets destroyed (and regardless of how this thing gets destroyed -- uncaught exception, unhandled signal, or normal exit). The type of `finalize()` (or more precisely of `&finalize()`) is `void(*)(lbann_comm*)`, so that's the second argument. It would also be correct to use `std::function<void(lbann_comm*)>`. I can change it to that if that would be easier to understand.

Reply:
I have updated to `std::function<void(lbann_comm*)>`.